### PR TITLE
rule-api update in preparation for release

### PR DESCRIPTION
**Pull request by anton-haubner-sonarsource**

### CODE REVIEW

File: python-checks/src/main/resources/org/sonar/l10n/py/rules/python/MISRA_C++_2008_profile.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (andreaguarino) [inline python-checks/src/main/resources/org/sonar/l10n/py/rules/python/MISRA_C++_2008_profile.json:1]: This file is unrelated to Python , so it shouldn't be commited

File: python-checks/src/main/resources/org/sonar/l10n/py/rules/python/MISRA_C++_2008_recommended_profile.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "MISRA C++ 2008 recommended",
```
Comment (andreaguarino) [inline python-checks/src/main/resources/org/sonar/l10n/py/rules/python/MISRA_C++_2008_recommended_profile.json:2]: This file is unrelated to Python , so it shouldn't be commited

File: python-checks/src/main/resources/org/sonar/l10n/py/rules/python/S2734.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "title": "\"__init__\" should not return a value",
   "type": "BUG",
-  "status": "ready",
+  "status": "deprecated",
```
Comment (andreaguarino) [inline python-checks/src/main/resources/org/sonar/l10n/py/rules/python/S2734.json:4]: @guillaume-dequenne-sonarsource @jean-jimbo-sonarsource is this deprecation intentional?

File: python-checks/src/main/resources/org/sonar/l10n/py/rules/python/Sonar_way_profile.json [hunk 4]
```diff
@@ -20,6 +20,7 @@
     "S1045",
     "S1066",
     "S1110",
+    "S1128",
```
Comment (andreaguarino) [inline python-checks/src/main/resources/org/sonar/l10n/py/rules/python/Sonar_way_profile.json:23]: This rule shouldn't be in SonarWay profile. We forgot to update RSPEC

File: python-checks/src/main/resources/org/sonar/l10n/py/rules/python/Sonar_way_profile.json [hunk 5]
```diff
@@ -201,6 +202,7 @@
     "S6560",
     "S6659",
     "S6660",
+    "S6661",
```
Comment (andreaguarino) [inline python-checks/src/main/resources/org/sonar/l10n/py/rules/python/Sonar_way_profile.json:205]: This rule shouldn't be in SonarWay profile. We forgot to update RSPEC

File: python-checks/src/main/resources/org/sonar/l10n/py/rules/python/S2734.json (hunk 3 above)
Comment (guillaume-dequenne-sonarsource) [reply python-checks/src/main/resources/org/sonar/l10n/py/rules/python/S2734.json:4]: It was apparently tagged as "superceded" by S935 2 years ago.
However, we did not implement S935 at the time, so it stayed "ready". I think this change has been triggered by the fact that S935 has now been implemented.
I assume the current implementation of S935 does not actually include `__init__` and the scope of S2734, right?
So probably we could keep the 2 rules and keep S2734 as a special case of S935: this seems to me the less disruptive approach.

Comment (anton-haubner-sonarsource) [reply python-checks/src/main/resources/org/sonar/l10n/py/rules/python/S2734.json:4]: Yes, the current implementation of S935 does not include `__init__`.
Also, the original formulation of the S935 rspec actually did not explicitly list the functions it targets.

Now, such a listing is included, as I merged it from the former S6658 rspec.

The original JIRA implementation ticket for S935 did list some special functions to be implemented, **including `__init__`**: https://sonarsource.atlassian.net/browse/SONARPY-519

 So for now, it is probably best, to keep S2734. When the time comes when work on [SONARPY-519](https://sonarsource.atlassian.net/browse/SONARPY-519) is continued, then it might be good to merge them.

File: python-checks/src/main/resources/org/sonar/l10n/py/rules/python/Sonar_way_profile.json (hunk 4 above)
Comment (anton-haubner-sonarsource) [reply python-checks/src/main/resources/org/sonar/l10n/py/rules/python/Sonar_way_profile.json:23]: I created a PR for this: https://github.com/SonarSource/rspec/pull/2772

File: python-checks/src/main/resources/org/sonar/l10n/py/rules/python/Sonar_way_profile.json (hunk 5 above)
Comment (anton-haubner-sonarsource) [reply python-checks/src/main/resources/org/sonar/l10n/py/rules/python/Sonar_way_profile.json:205]: I created a PR for this: https://github.com/SonarSource/rspec/pull/2772

File: python-checks/src/main/resources/org/sonar/l10n/py/rules/python/Sonar_way_profile.json (hunk 4 above)
Comment (anton-haubner-sonarsource) [reply python-checks/src/main/resources/org/sonar/l10n/py/rules/python/Sonar_way_profile.json]: A PR for rspec has been created: https://github.com/SonarSource/rspec/pull/2773

Comment (anton-haubner-sonarsource) [reply python-checks/src/main/resources/org/sonar/l10n/py/rules/python/Sonar_way_profile.json]: Also, it has been fixed manually now for sonar-python in this branch. Same for S6661

**Review by andreaguarino — APPROVED** [review]
LGTM

File: python-checks/src/main/resources/org/sonar/l10n/py/rules/python/S2734.json (hunk 3 above)
Comment (anton-haubner-sonarsource) [reply python-checks/src/main/resources/org/sonar/l10n/py/rules/python/S2734.json]: I created an rspec PR to revert the deprecation of S2734 for now, until S935 is extended to cover more special functions: https://github.com/SonarSource/rspec/pull/2774
